client/packages/purchasing/src/purchase_order/DetailView/LineEdit/PurchaseOrderLineEdit.tsx

```diff
@@ -55,282 +60,304 @@ export const PurchaseOrderLineEdit = ({
+              value={draft?.requestedNumberOfUnits}
+              disabled={true}
+              isVerticalScreen={isVerticalScreen}
+              decimalLimit={2}
+            />
+            {!canEditRequestedQuantity && (
               <NumInputRow
-                label={t('label.total-cost')}
-                value={
-                  draft
-                    ? (draft.pricePerUnitAfterDiscount ?? 0) *
-                      (draft.adjustedNumberOfUnits ??
-                        draft?.requestedNumberOfUnits ??
-                        0)
-                    : 0
-                }
-                disabled
+                label={t('label.adjusted-units')}
+                value={draft?.adjustedNumberOfUnits ?? undefined}
+                disabled={true}
                 isVerticalScreen={isVerticalScreen}
+                decimalLimit={2}
               />
-            </>
-          ) : null
-        }
-        Right={
-          showContent ? (
-            <>
-              {dateInput(
-                'label.requested-delivery-date',
-                draft?.requestedDeliveryDate,
-                value => update({ requestedDeliveryDate: value }),
-                {
-                  disabled: isFieldDisabled(status, StatusGroup.AfterConfirmed),
-                }
-              )}
-              {dateInput(
-                'label.expected-delivery-date',
-                draft?.expectedDeliveryDate,
-                value => update({ expectedDeliveryDate: value }),
-                {
-                  disabled: isFieldDisabled(status, StatusGroup.AfterSent),
-                }
-              )}
-              {multilineTextInput(
-                'label.comment',
-                draft?.comment || '',
-                value => update({ comment: value })
-              )}
-              {multilineTextInput('label.notes', draft?.note || '', value =>
-                update({ note: value })
-              )}
-            </>
-          ) : null
-        }
-      />
-    </>
+            )}
+            <NumInputRow
+              label={t('label.price-per-pack-before-discount')}
+              value={getCurrencyValue(draft?.pricePerPackBeforeDiscount)}
+              disabled={
+                disabled || isFieldDisabled(status, StatusGroup.AfterConfirmed)
+              }
+              isVerticalScreen={isVerticalScreen}
+              onChange={(value: number | undefined) => {
+                const adjustedPatch = calculatePricesAndDiscount(
+                  'pricePerPackBeforeDiscount',
+                  { ...draft, pricePerPackBeforeDiscount: value }
+                );
+                update(adjustedPatch);
+              }}
+              decimalLimit={options.precision}
+              endAdornment={options.symbol}
+            />
+            <NumInputRow
+              label={t('label.discount-percentage')}
+              value={draft?.discountPercentage ?? undefined}
```

Comment:
This isn't handling NaN

```suggestion
              value={draft?.discountPercentage || 0}
```